bpf/lib/nodeport_egress.h

```diff
@@ -701,10 +717,12 @@ handle_nat_fwd(struct __ctx_buff *ctx, __u32 cluster_id, __u32 src_id,
 #endif /* ENABLE_IPV4 */
 #ifdef ENABLE_IPV6
 	case bpf_htons(ETH_P_IPV6):
-		ret = invoke_traced_tailcall_if(__or(__and(is_defined(ENABLE_IPV4),
-							   is_defined(ENABLE_IPV6)),
-						     __and(is_defined(ENABLE_HOST_FIREWALL),
-							   is_defined(IS_BPF_HOST))),
+		ret = invoke_traced_tailcall_if(__or3(__and(is_defined(ENABLE_IPV4),
+							    is_defined(ENABLE_IPV6)),
+						      __and(is_defined(ENABLE_HOST_FIREWALL),
+							    is_defined(IS_BPF_HOST)),
+						      __and(is_defined(ENABLE_EGRESS_GATEWAY_COMMON),
```

Comment:
This currently throws a checkpatch error, however, the ipv4 code looks exactly the same so I thought we might be able to ignore that given that checkpatch is not required to pass.

Reply:
sgtm

Reply:
@ysksuzuki discussion of why we accepted the checkpatch failure.